### PR TITLE
Improve info command output when a stanza is specified but missing.

### PR DESCRIPTION
**Pull request by pgstef**
Idea based on issue https://github.com/pgbackrest/pgbackrest/issues/1199.

Output would give:

```
$ pgbackrest_build info --output=json
[]

$ pgbackrest_build info --output=json --stanza=foo
[{"backup":[],"db":[],"name":"foo","status":{"code":1,"lock":{"backup":{"held":false}},"message":"missing stanza path"}}]

$ pgbackrest_build info 
No stanzas exist in the repository.

$ pgbackrest_build info --stanza=foo
stanza: foo
    status: error (missing stanza path)
```

It's a pretty minor code change and I think it doesn't break the actual json format used... This can be a better output than the empty one.

### CODE REVIEW

File: test/src/module/command/infoTest.c [hunk 1]
```diff
@@ -42,6 +42,37 @@ testRun(void)
...
+                    "\"status\":{"
+                        "\"code\":1,"
+                        "\"lock\":{\"backup\":{\"held\":false}},"
+                        "\"message\":\"missing stanza path\""
+                        "}"
+                "}"
+            "]",
+            "json - empty repo, stanza option specified");
+
+
```
Comment (cmwshang) [inline test/src/module/command/infoTest.c:66]: Extra empty line not needed.

**Review by cmwshang — COMMENTED** [review]
Overall the change looks good. I am just concerned that anyone out there relying on the empty format for their tests or monitoring might not be happy. Has anyone else complained? We might want to solicit a bit more feedback...

File: test/src/module/command/infoTest.c (hunk 1 above)
Comment (pgstef) [reply test/src/module/command/infoTest.c]: removed, thanks

**Comment by pgstef** [comment]
Currently I think that's the only one who reported an issue. I would find a bit strange to rely on "empty" for monitoring purpose though.

This new behaviour seems more understandable at first sight:
- asking for a specific stanza, not there -> error
- asking for the complete info, nothing there -> empty

From a monitoring point of view, I find it better to rely on error codes rather than on empty return (which could mean anything).
Regarding tests, if the question is "does my stanza exist? if not, create it", I think it's better to use an error code rather then an empty output. Although, the text output isn't empty.

Solicit feedback might make sense but how can we actually do that ?

**Comment by cmwshang** [comment]
> From a monitoring point of view, I find it better to rely on error codes rather than on empty return (which could mean anything).

Good point.

**Review by cmwshang — APPROVED** [review]

**Review by dwsteele — APPROVED** [review]

**Pull request merged**